Add jsx-no-literals in warning mode

arski opened the pull request:
@udemy/team-f can you advise on the right steps sequence for this.. I presume merging and publishing here won't affect anything until we upgrade this package in website-django.. then again thinking of adding as warning first anyway, skipping the rule on certain paths when adding it (e.g. tests, examples) and then having people fix their code before upgrading to error status. wdyt?

PS. Checked Inanc's comment and yea, just going with the popular version for now until it becomes clear that we need to adjust it.

arski committed:
feat: Introduce jsx-no-literals

arski commented:
reference https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-literals.md

jjinux commented:
> I presume merging and publishing here won't affect anything until we upgrade this package in website-django..

Right.

>  then again thinking of adding as warning first anyway, skipping the rule on certain paths when adding it (e.g. tests, examples) and then having people fix their code before upgrading to error status. wdyt?

We never use warnings since they're generally just ignored. We generally fix everyone's code for them.

arski commented:
@jjinux all right, changed. I do like to fix some code myself, but also want to annoy people for forgetting this.. will see :)

jjinux reviewed (approved):
I'm sorry to ask a stupid question, but did you verify that this works out okay in practice for our codebase? I.e. it doesn't cause us to do silly things, does it?

arski commented:
Not stupid at all, I did test the rule on our codebase and it does what it says it will - highlight any string literals in between jsx tags, from what I observed (except for style guide, examples, tests), all of them are correctly highlighted as strings missed to be marked for translation (perhaps intentionally, but that's for code owners to eslint-ignore-rule then).

What are the next steps with this? let's chat in person in your morning if you don't mind. I would really prefer not to have to apply the curly braces rule across the site that @kevinzang mentioned is also pending from js-tooling :)

jjinux commented:
The curly brace rule probably shouldn't have been merged before we were ready to apply it to our codebase because it's just an accident waiting to happen ;) It's like merging to master without doing a deploy. @jkang1980, perhaps you can back it out, and leave it as a PR ready to be merged?

@arski, can you start updating website-django so that it'll comply with what you're trying to merge?

Then, after @jkang1980 is done, and you have website-django in a good shape, you can both merge your change adding the new rule to js-tooling and then merge a change to website-django to use the new version of the library.

jjinux commented:
Oops, @kngroo. ^

arski commented:
Hey can i point website-django at this commit somehow so i can test? I tried but it doesnt seem to work.. Its an interestimg repo with multiple packages..

kevinzang commented:
@arski I'm not sure how to do that, but as an alternative, you might find it easier to edit node_modules directly in website-django, and then after confirming things work, copy your changes over to js-tooling.

inancsevinc commented:
@arski in the past, what I did was to modify https://github.com/udemy/website-django/blob/master/static/src/.eslintrc.js in the website-django and add the rule I'm adding to this repo directly in that file.

arski commented:
aha that works of course! i was looking at static/.eslintrc and that was horribly empty, but the one in src/ is perfect :)

arski commented:
btw did you know there are quite a few react/jsx-no-bind errors around our code - is that something you can fix with an auto-fixer?

cansin commented:
How come CI won't catch any existing errors in the codebase though?

arski commented:
dunno, no idea?

cansin commented:
@jjinux ? :D

jjinux commented:
That's really weird. I'll create a card.

Commit:
Publish

 - babel-polyfill-udemy-website@0.6.6
 - babel-preset-udemy-website@0.8.1
 - eslint-config-udemy-react-addons@3.2.0
 - eslint-config-udemy-website@4.1.2
